woocommerce/woocommerce
# Try: Product Form CPT View Render

### Changes proposed in this Pull Request:

This PR is a continuation of https://github.com/woocommerce/woocommerce/pull/45969 and follows a more blocks-like approach of separating the `Edit` and `Save` render of blocks.  The difference between traditional blocks that this PR introduces is the concept of a `View` that allows injection of React components.

* Register and use a `View` component for rendering blocks that require React in the view
* Create a util to render a tree of components from an HTML string
* Fallback to `Edit` component in Woo product blocks when no `View` is present
<img width="1037" alt="Screenshot 2024-04-02 at 2 17 19 PM" src="https://github.com/woocommerce/woocommerce/assets/10561050/4647997e-f843-4804-a9a1-3207c2fb4010">





### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Navigate to Product Form -> Add new
2. Create a title for your form and add some fields (only checkbox, regular price, name, and section have been updated to work in this PR)
3. Save your form and optionally create another
4. Navigate to Products -> Add new
5. See your form has been used to populate the fields in the product editor.  Note that fields with registered `View` components inject their`View` components while those without `View` (checkbox) continue to show the `Edit` component
6. If you created multiple forms, switch between forms using the select dropdown



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
